R/misc.R

```diff
@@ -21,13 +21,21 @@ get_rhs_vars <- function(formula, data, no_lhs = FALSE) {
   ## or should it? what about Y ~ log(x)?
   ## Answer: when called from `form2args`, the function
   ## `inline_check` stops when in-line functions are used.
-  data_info <- attr(model.frame(formula, data[1, ]), "terms")
-  response_info <- attr(data_info, "response")
-  predictor_names <- names(attr(data_info, "dataClasses"))
-  if (length(response_info) > 0 && all(response_info > 0)) {
-    predictor_names <- predictor_names[-response_info]
+
```

Comment:
We are allowed to do this refactor because `get_rhs_vars()` is only ever called after `inline_check()` in `form2args()`

https://github.com/tidymodels/recipes/blob/main/R/recipe.R#L233-L241

the the question in essence because:

1. if formula contains a `.`, use "all names" otherwise use `all.names()` to pull them out.
1. pull names out of lhs
2. take the diff if needed